src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -24,9 +24,15 @@ internal sealed class EnumConverter<T> : JsonConverter<T>
 
         private readonly JsonNamingPolicy? _namingPolicy;
 
-        private readonly ConcurrentDictionary<ulong, JsonEncodedText> _nameCache;
+        /// <summary>
+        /// Holds a mapping from enum value to text that might be formatted with <see cref="_namingPolicy" />.
+        /// </summary>
+        private readonly ConcurrentDictionary<ulong, JsonEncodedText> _nameCacheForWriting;
```

Comment:
is converter the right place for this? could converter be potentially re-used in place with different naming policy? Would it make sense to move this to JsonTypeInfo and cache all possible values beforehand (i.e. in Configure method) and then use regular Dictionary since I believe just lookup is thread-afe?

Reply:
> could converter be potentially re-used in place with different naming policy?

I don't believe so.

> Would it make sense to move this to JsonTypeInfo and cache all possible values beforehand

The enum converter might be applied to enum types or specific properties. Placing on enum `JsonTypeInfo`s would satisfy only the former.

Reply:
Might be good to still considering pre-caching all values in the constructor since that won't require to use concurrent dictionary but consider this feedback optional

Reply:
also another alternative to put the caching both on JsonPropertyInfo and JsonTypeInfo but if JsonPropertyInfo uses JsonTypeInfo's dictionary then you can re-use the same dictionary but assuming JsonConverter is unique per JsonTypeInfo/JsonPropertyInfo consider this optional

Reply:
hmm.. seems prepopulating might not work due to combinations of flags